src/Features/CSharp/Portable/Completion/KeywordRecommenders/BoolKeywordRecommender.cs

```diff
@@ -44,5 +45,10 @@ protected override bool IsValidContext(int position, CSharpSyntaxContext context
                     canBePartial: false,
                     cancellationToken: cancellationToken);
         }
+
+        protected override bool ShouldPreselect(CSharpSyntaxContext context, CancellationToken cancellationToken)
+        {
+            return context.InferredTypes.Any(t => t.SpecialType == SpecialType.System_Boolean);
```

Comment:
There's a lot of repeated code.  Have a subclass that has this logic, and hte base types pass in something like "SpecialType.System_Boolean".